bitcoin/src/crypto/sighash.rs

```diff
@@ -512,7 +512,7 @@ impl TapSighashType {
             0x81 => AllPlusAnyoneCanPay,
             0x82 => NonePlusAnyoneCanPay,
             0x83 => SinglePlusAnyoneCanPay,
-            x => return Err(InvalidSighashTypeError(x.into())),
+            x => return Err(InvalidSighashTypeError(x as u32)), // Cast ok, u8 -> u32
```

Comment:
Maybe we should have something like `internals::cast::u8_to_u32` to reduce these explanations and possibly allow clippy lints.